src/utilities/device_utils.c

```diff
@@ -5,6 +5,7 @@
  * SPDX-License-Identifier: (Apache-2.0 OR MIT)
  ******************************************************************************/
 
+#include <math.h>
```

Comment:
@liruipeng, make sure that whenever `_hypre_onedpl.hpp` is included, that file should be included first. Putting any includes before it may break the sycl build.

Reply:
@waynemitchell OK Thanks!

Reply:
I moved `math.h` after it. Is it OK now?

Reply:
Yes, should be fine so long as it comes after the include of `_hypre_onedpl.hpp`. I've moved it in my local version of your branch as well.